### PR TITLE
fix(apim): error port name too long

### PR DESCRIPTION
**Pull request by ThibaudAV**
**Issue**

https://github.com/gravitee-io/issues/issues/8345

Followup of https://github.com/gravitee-io/helm-charts/pull/292

**Description**

Crop all ports names to ensure they have a correct size

**Additional context**

### CODE REVIEW

**Review by mariano-fernandez — APPROVED** [review]

**Review by leleueri — COMMENTED** [review]
Can you add unit tests on AM please and add an entry into the Changelogs + Chart.yml to inform about the change ?
In addition, name limitation look to be restricted to the POD spec, not in the sevice spec, but for consistency against products, I'm ok to apply it on AM too.

**Comment by mariano-fernandez** [comment]
> Can you add unit tests on AM please and add an entry into the Changelogs + Chart.yml to inform about the change ?

There is already a unit-test for cockpit, but can we get the `changelog + chart.hml` part ? 😬

**Comment by ThibaudAV** [comment]
@leleueri I didn't really find any doc on this subject.
But if it's only for pods then I leave as before on services ? or limit to 63
And add limit only when needed

File: am/templates/gateway/gateway-service.yaml [hunk 1]
```diff
@@ -31,7 +31,7 @@ spec:
       {{ end }}
       {{ end }}
       {{- end }}
-      name: {{ .Values.gateway.name }}
+      name: {{ .Values.gateway.name | trunc 15 | trimSuffix "-" }}
```
Comment (leleueri) [inline am/templates/gateway/gateway-service.yaml:34]: On Service resource, names are limited to 63 chars

**Comment by remisultan** [comment]
ditto all of the above for AE with `Unit tests + Charts.yml`  change

File: am/templates/gateway/gateway-service.yaml [hunk 2]
```diff
@@ -43,7 +43,7 @@ spec:
       appProtocol: http
       {{ end }}
       {{- end }}
-      name: {{ .Values.gateway.name }}-technical
+      name: {{ printf "%s-%s" (.Values.gateway.name | trunc 5 | trimSuffix "-") "technical" }}
```
Comment (leleueri) [inline am/templates/gateway/gateway-service.yaml:46]: On Service resource, names are limited to 63 chars so here the trunc should be done at 53 chars in order to be able to append "-technical"

File: am/templates/api/api-service.yaml [hunk 3]
```diff
@@ -31,7 +31,7 @@ spec:
       {{ end }}
       {{ end }}
       {{- end }}
-      name: {{ .Values.api.name }}
+      name: {{ .Values.api.name | trunc 15 | trimSuffix "-" }}
```
Comment (leleueri) [inline am/templates/api/api-service.yaml:34]: On Service resource, names are limited to 63 chars

File: am/templates/api/api-service.yaml [hunk 4]
```diff
@@ -43,7 +43,7 @@ spec:
       appProtocol: http
       {{ end }}
       {{- end }}
-      name: {{ .Values.api.name }}-technical
+      name: {{ printf "%s-%s" (.Values.api.name | trunc 5 | trimSuffix "-") "technical" }}
```
Comment (leleueri) [inline am/templates/api/api-service.yaml:46]: On Service resource, names are limited to 63 chars so here the trunc should be done at 53 chars in order to be able to append "-technical"

File: am/templates/ui/ui-service.yaml [hunk 5]
```diff
@@ -27,7 +27,7 @@ spec:
       appProtocol: http
       {{ end }}
       {{- end }}
-      name: {{ .Values.ui.name }}
+      name: {{ .Values.ui.name | trunc 15 | trimSuffix "-" }}
```
Comment (leleueri) [inline am/templates/ui/ui-service.yaml:30]: On Service resource, names are limited to 63 chars

**Review by leleueri — CHANGES_REQUESTED** [review]

**Comment by ThibaudAV** [comment]
Following discussion with @leleueri . The limit is kept at 63 for services and 15 deployment. as the one imposed by k8s

**Review by leleueri — APPROVED** [review]

**Review by gaetanmaisse — APPROVED** [review]

**Pull request merged**